File: GAF/java/src/DataHandler.java
public class DataHandler {
    
    private float[] voltage;  // Potential (V)
    private float[] current;  // Current (A)
    private int dataPoints;
    
    // Min/Max values for normalization
    private float maxVoltage;
    private float minVoltage;
    private float maxCurrent;
    private float minCurrent;
    
    public DataHandler(float[] voltage, float[] current) {
        this.voltage = voltage;
        this.current = current;
        this.dataPoints = voltage.length;
        
        // Calculate min/max values
        calculateMinMax();
    }
    
    private void calculateMinMax() {
        maxVoltage = Float.MIN_VALUE;
        minVoltage = Float.MAX_VALUE;
        maxCurrent = Float.MIN_VALUE;
        minCurrent = Float.MAX_VALUE;
        
        for (int i = 0; i < dataPoints; i++) {
            if (voltage[i] > maxVoltage) maxVoltage = voltage[i];
            if (voltage[i] < minVoltage) minVoltage = voltage[i];
            if (current[i] > maxCurrent) maxCurrent = current[i];
            if (current[i] < minCurrent) minCurrent = current[i];
        }
    }
    
    // Getters
    public float[] getVoltage() {
        return voltage;
    }
    
    public float[] getCurrent() {
        return current;
    }
    
    public int getDataPoints() {
        return dataPoints;
    }
    
    public float getMaxVoltage() {
        return maxVoltage;
    }
    
    public float getMinVoltage() {
        return minVoltage;
    }
    
    public float getMaxCurrent() {
        return maxCurrent;
    }
    
    public float getMinCurrent() {
        return minCurrent;
    }
    
    public void printDataInfo() {
        System.out.println("  Data Points: " + dataPoints);
        System.out.println("  Voltage Range: [" + minVoltage + ", " + maxVoltage + "]");
        System.out.println("  Current Range: [" + minCurrent + ", " + maxCurrent + "]");
    }
}

File: GAF/python/data_handler.py
import numpy as np

class DataHandler:
    def __init__(self, voltage, current):
        self.voltage = np.asarray(voltage, dtype=np.float32)
        self.current = np.asarray(current, dtype=np.float32)
        self.dataPoints = len(self.voltage)

        self._calculateMinMax()
    
    def _calculateMinMax(self):
        self.maxVoltage = float(np.max(self.voltage))
        self.minVoltage = float(np.min(self.voltage))
        self.maxCurrent = float(np.max(self.current))
        self.minCurrent = float(np.min(self.current))
    
    def getVoltage(self):
        return self.voltage
    
    def getCurrent(self):
        return self.current
    
    def getDataPoints(self):
        return self.dataPoints
    
    def getMaxVoltage(self):
        return self.maxVoltage
    
    def getMinVoltage(self):
        return self.minVoltage
    
    def getMaxCurrent(self):
        return self.maxCurrent
    
    def getMinCurrent(self):
        return self.minCurrent
    
    def printDataInfo(self):
        print(f"  Data Points: {self.dataPoints}")
        print(f"  Voltage Range: [{self.minVoltage} , {self.maxVoltage}]")
        print(f"  Current Range: [{self.minCurrent} , {self.maxCurrent}]")